### ISSUE TITLE
HGIG / Black-Level / PC mode no longer available

### ISSUE THREAD

**Issue by pho084** [Cathwyler/MagicRemoteService#21]
first of all thank you for this awesome application! Controls work super smooth on a rooted LG C9

the only thing that is not working as i would have hoped is the source/input setting.

when configuring HDMI inputs via LG's device connector you can choose "PC" and set up the connected PC accordingly.
this enables among others Chroma 4:4:4:4 and some HDR such as HGIG mode (dynamic tonemapping). It also let's you control / adjust black levels (in "picture settings" and "expert settings" respectively)

now, since MagicRemoteService creates a "new" HDMI input device the settings from PC mode do not carry over. Instead the HDMI input is treated like a "normal" HDMI device. This means losing HGIG, BlackLevel adjustment and probably some other features.

long story short: it would be great if MagicRemoteService could be configured to actually emulate a "PC" device HDMI input the way LG understands it :)

thanks again!

**Comment by lDeKul** [comment]
I do have the same "issue" using the newest prerelese 1.2.3.0
I would be truly grateful if the input could be fully recognized as a PC.

Thank you again for this great piece of Software.

**Comment by Cathwyler** [comment]
Hi,

Thanks for you feedback.

I actually use a [non oficial addDevice API](https://www.webosbrew.org/pages/luna-service-comwebosserviceeim.html) to add a TV inputs and these additional inputs are identified as a streaming service. ```[string] type - missing description (supported values: MVPD_IP, MVPD_RF)```

But there may be a workaround for rooted TV. Overwriting the official HDMI input may be the solution. You probably need to overwrite all package name com.cathwyler.magicremoteservice by com.webos.app.hdmi1 in the source file and build the app.
Be really careful and save all TV files you need to save before overwriting.

An other possibility is that the addDevice API type "HDMI1, HDMI-1, HDMI_1, ...." instead of "MVPD_IP, MVPD_RF" works only after deleting the official hdmi package.

On my side, I don't wanna risk to root or brick my TV.
But if anyone have the clue and can share me any solution i can add an option to build a HDMI overwriting package.

**Comment by pho084** [comment]
thank you for looking into this.

i see. that sounds like it could break (i.e. "brick") more than it could fix if not done properly :D 

i lack the expertise here. thanks nevertheless

**Comment by Cathwyler** [comment]
Hi,

I just added a new overlay option over native hdmi app. That can maybe fix your issue in this pre-release [1.2.4.0](https://github.com/Cathwyler/MagicRemoteService/releases/tag/1.2.4.0)

Not that this option is fully experimental with plenty of unofficial webOS option. It works on webOS 6.0.
Let me know if it works fine.

Regards

**Comment by Cathwyler** [comment]
Na don't worry.
I'm only using unofficial API, i'm not replacing any system or native files.
If it's not working just uncheck the new option and install again on TV 

I just fixe a Wake on Lan bug use 1.2.4.1 version instead of 1.2.4.0

**Comment by pho084** [comment]
sorry for the dumb question but just to make double sure considering update process:

1) kill magicremote service running in background via task manager
2) copy/overwrite contents of archive of latest version to original magicremote folder
3) turn on tv
4) run magicremote.exe on PC and click on "install on tv" to reinstall ipk

or do i have to manually remove the app from the TV first?
thanks

**Comment by Cathwyler** [comment]
You don't need to uninstall the app before reinstall on TV.
I added the version number in the latest release to be sure the right one is running.

You need to check the new overlay option before reinstall on TV if you want to test it.
With this option, checked the app will now run in overlay mode above the native hdmi app input.
That should fix the HGIG / Black Level / PC mode because the native app handle it properly.

**Comment by pho084** [comment]
perfect. thanks for the reply. 

but apart from that my step-by-step update guide should work if i understand correctly?

**Comment by Cathwyler** [comment]
Yep it should work

**Comment by pho084** [comment]
finally got around to updating 

initial tests: It's working! Overlay mode fixes all my issues with HGIG etc. Gsynch also still working

great work and again thanks a million @Cathwyler 

p.s.: one thing to note: during installation Windows Defender picked up the .js file and blocked it effectively aborting installation on TV during the process. this caused the TV to "panic". Power Cycling the TV and disabling Windows Defender for the installation solved the problem (now i just hope it really was a false positive :D)

**Comment by Cathwyler** [comment]
I think its a false.
But yeah an administrator executable file launching service and creating script without any certificat will trigger more than one antivirus.

But to be sure, you can verify that there no injected script in the blocked js file.
